pkg/types/chain_reader.go

```diff
@@ -0,0 +1,76 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc/status"
+)
+
+// Errors exposed to product plugins
+
+type errChainReader string
+
+func (e errChainReader) Error() string { return string(e) }
+
+const (
+	ErrInvalidType   = errChainReader("invalid type")
+	ErrFieldNotFound = errChainReader("field not found")
+)
+
+func UnwrapClientError(err error) error {
+	if s, ok := status.FromError(err); ok {
+		return errChainReader(s.String())
+	}
+	return err
+}
+
+// Errors used only by relay plugins
+
+type ErrorChainReaderUnsupported struct{}
+
+func (e ErrorChainReaderUnsupported) Error() string {
+	return "ChainReader is not supported by the relay"
+}
+
+type ErrorNoChainReaderInJobSpec struct{}
+
+func (e ErrorNoChainReaderInJobSpec) Error() string {
+	return "There is no ChainReader configuration defined in the job spec"
+}
+
+type ErrorChainReaderInvalidConfig struct{}
```

Comment:
Why is it important to have an error specialized for chain reader? The callers already know what they are configuring. What about:
```suggestion
var ErrInvalidConfig = errors.New("invalid configuration")
```